lasagne/layers/recurrent.py

```diff
@@ -166,6 +167,22 @@ class CustomRecurrentLayer(MergeLayer):
     ...     lasagne.layers.InputLayer((None, n_hid)), n_hid)
     >>> l_rec = lasagne.layers.CustomRecurrentLayer(l_in, l_in_hid, l_hid_hid)
 
+    The CustomRecurrentLayer can also support "convolutional recurrence", as is
+    demonstrated below.
+
+    >>> n_batch, n_steps, n_channels, width, height = (2, 3, 4, 5, 6)
+    >>> n_out_filters = 7
+    >>> filter_shape = (3, 3)
+    >>> l_in = lasagne.layers.InputLayer(
+    ...     (n_batch, n_steps, n_channels, width, height))
+    >>> l_in_to_hid = lasagne.layers.Conv2DLayer(
+    ...     lasagne.layers.InputLayer((None, n_channels, width, height)),
+    ...     n_out_filters, filter_shape, pad='same')
+    >>> l_hid_to_hid = lasagne.layers.Conv2DLayer(
+    ...     lasagne.layers.InputLayer((None, n_out_filters, width, height)),
```

Comment:
What about using `l_in_to_hid.output_shape` instead of `(None, n_out_filters, width, height)` here? That's a bit more flexible (i.e., it will still work with different padding for `l_in_to_hid` or different convolution layer implementations (some don't do `pad='same'` correctly for even filter sizes, they actually do `pad='half'` instead)).